### PR TITLE
Autoplay on mobile in muted state if autostart is set to true

### PR DESCRIPTION
**Pull request by egreaves**
### Changes proposed in this pull request:
Set the `muted` and `autoplay` attributes on mobile when `autostart = true` so browsers autoplay videos when they are in the viewport. 

Fixes #
JW7-2962

### CODE REVIEW

**Comment by egreaves** [comment]
test this please

**Comment by egreaves** [comment]
test this please

**Comment by dannyfinks** [comment]
test this please

**Comment by dannyfinks** [comment]
test this please

**Comment by dannyfinks** [comment]
@egreaves it seems like if we also have repeat set to true the stream will not repeat on mobile. The player will go to idle state. Also if possible to investigate if we can autostart while the player is in the viewport. http://player-pr-test-jenkins.longtailvideo.com/builds/3534/archive/test/squash/bootstrap.html?primary=html5&edition=ads&feature=options_repeat

**Comment by egreaves** [comment]
test this please

**Comment by dannyfinks** [comment]
@egreaves it seems like not just repeat is impacted by this issue, but playlists as well. Once the autostarted first playlist item completes the player will go to idle state showing the poster image of the second playlist item

**Comment by dannyfinks** [comment]
@egreaves can you see on an android chrome device if this page will page with autostart players will crash for you? http://player-pr-test-jenkins.longtailvideo.com/builds/3582/archive/test/squash/bootstrap.html?primary=html5&edition=ads&feature=Autostart/autostart_hls_dash I am not able to reproduce in release / develop, but I can in the MAUI branch. I can't reproduce on iPhone only on my Android Chrome tablet.

File: src/js/providers/html5.js [hunk 1]
```diff
@@ -448,6 +451,11 @@ define([
         }
 
         function _play() {
+            if (utils.isIOS() && _videotag.readyState !== 4) {
```
Comment (hghazzi) [inline src/js/providers/html5.js]: Does readyState have a Enum we can use instead?

File: src/js/providers/html5.js [hunk 2]
```diff
@@ -145,7 +145,9 @@ define([
             _audioTracks = null,
             _currentAudioTrackIndex = -1,
             _activeCuePosition = -1,
-            _visualQuality = { level: {} };
+            _visualQuality = { level: {}},
```
Comment (hghazzi) [inline src/js/providers/html5.js]: Nitpick, but can you fix the whitespace here.

File: src/js/controller/model.js [hunk 3]
```diff
@@ -40,11 +40,10 @@ define([
...
             });
 
-            // Mobile doesn't support autostart
-            // This check should be replaced with something that detects whether the current system
-            // requires a user interaction to start playback
-            if (utils.isMobile() && !config.mobileSdk) {
-                this.set('autostart', false);
+            // Videos autostart on mobile when the `muted` and `autoplay` attributes are set.
+            // Mute the player on a mobile device if autostart is set to true
+            if (this.get('autostart') && utils.isMobile() && !this.get('mobileSdk')) {
```
Comment (hghazzi) [inline src/js/controller/model.js]: Might want to movie this into its own function, so the logic is shared between here and below.

File: src/js/providers/html5.js (hunk 1 above)
Comment (egreaves) [inline src/js/providers/html5.js]: We'd have to create it.

File: src/js/controller/model.js [hunk 4]
```diff
@@ -219,6 +218,15 @@ define([
...
             _provider.volume(_this.get('volume'));
             _provider.mute(_this.get('mute'));
+
+            if (_provider.removeAutoplayAttributes) {
+                _provider.removeAutoplayAttributes();
+            }
+
+            // set autoplay attributes if on a mobile device and autostart is true
+            if (this.autoStartOnMobile() && _provider.setAutoplayAttributes) {
+                    _provider.setAutoplayAttributes();
```
Comment (hghazzi) [inline src/js/controller/model.js]: Fix tabbing

File: src/js/providers/html5.js [hunk 5]
```diff
@@ -448,6 +451,12 @@ define([
         }
 
         function _play() {
+            // Wait until the canplay event (readyState === 4) fires to play on iOS.
+            // Otherwise, calling play will result in an AbortError
```
Comment (hghazzi) [inline src/js/providers/html5.js]: Could you also mention that 4 is considered HAVE_ENOUGH_DATA in the spec.

**Pull request merged**